sympy/tensor/array/expressions/from_to_einsum.py

```diff
@@ -0,0 +1,35 @@
+from collections import Counter, defaultdict
+
+from sympy.tensor.array.expressions import ArrayTensorProduct, ArrayContraction, ArrayDiagonal, PermuteDims
+
+
+def einsum_to_sympy_array(path, *args):
```

Comment:
This function is very hard to read. Can you at least add some blank lines? 

I would split out the string path parsing into a separate function, especially since paths can already be represented as lists of tuples natively in np.einsum.

Reply:
Also the function needs a docstring.